demos/custom-tabs-headers/src/main/java/com/google/androidbrowserhelper/demos/customtabsheaders/MainActivity.java

```diff
@@ -0,0 +1,123 @@
+/*
+ *    Copyright 2020 Google LLC
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.google.androidbrowserhelper.demos.customtabsheaders;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.browser.customtabs.CustomTabsCallback;
+import androidx.browser.customtabs.CustomTabsClient;
+import androidx.browser.customtabs.CustomTabsIntent;
+import androidx.browser.customtabs.CustomTabsService;
+import androidx.browser.customtabs.CustomTabsServiceConnection;
+import androidx.browser.customtabs.CustomTabsSession;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Browser;
+import android.widget.Toast;
+
+public class MainActivity extends AppCompatActivity {
+    private static final Uri URL = Uri.parse("https://padr31.github.io/");
+
+    private CustomTabsSession mSession;
+    private CustomTabsServiceConnection mConnection;
+
+    private boolean connectionValidated = false;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        // Set up a callback that launches the cross origin intent after session validated.
+        CustomTabsCallback callback = new CustomTabsCallback() {
+            @Override
+            public void onRelationshipValidationResult(int relation,
+                                                       @NonNull Uri requestedOrigin,
+                                                       boolean result,
+                                                       @Nullable Bundle extras) {
+                // Can launch custom tabs intent after session was validated as the same origin.
+                connectionValidated = true;
```

Comment:
What I would do is make `btn_extra` disabled by default, and then enable it once validation is done.